File: ABC/ABC361/C/main.py
n, k = map(int, input().split())
a = sorted(list(map(int, input().split())))

l = n - k
res = a[-1] - a[0]
for i in range(k + 1):
    res = min(res, a[i + l - 1] - a[i])

print(res)

File: ABC/ABC361/C/main.cc
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int n, k;
  cin >> n >> k;
  vector<int> a(n);
  for (auto &e : a) cin >> e;
  sort(begin(a), end(a));

  int len = n - k;
  int res = a.back() - a[0];
  for (int i = 0; i <= k; ++i) {
    res = min(res, a[i + len - 1] - a[i]);
  }

  cout << res << endl;
  return 0;
}